pkg/utils/nodes.go

```diff
@@ -22,17 +22,17 @@ func GetNodeWithName(r client.Reader, nodeName string) (*corev1.Node, error) {
 }
 
 // IsNodeNameValid returns an error if nodeName doesn't match any node name int the cluster, otherwise a nil
-func IsNodeNameValid(r client.Reader, nodeName string) (bool, error) {
-	_, err := GetNodeWithName(r, nodeName)
+func IsNodeNameValid(r client.Reader, nodeName string) (*corev1.Node, bool, error) {
```

Comment:
I think also the name of the function should be changed, since it does not only return if the node is valid, but the node itself (like "GetNodeWithName")